### PR TITLE
H1rono/assimp fetch

### LINKED ISSUE #220: Support loading online mesh with assimp
Currently, meshes starting with `http:`/`https:` are not supported with assimp.

https://github.com/openrr/urdf-viz/blob/9cf51a756a748a78644f3923756970e2bf2ac7af/src/mesh.rs#L104-L107

This is because the assimp-sys API requires the path to the local mesh file.

https://github.com/openrr/urdf-viz/blob/9cf51a756a748a78644f3923756970e2bf2ac7af/src/mesh.rs#L26

By downloading the file, saving it as a temporary file, and passing the path to it to assimp-sys, it should be possible to support meshes starting with `http:`/`https:` with assimp.

### PR DESCRIPTION
**Pull request by H1rono**
close #220 
Tested with `cargo run -- https://github.com/openrr/urdf-viz/raw/main/sample.urdf` on macOS.

### CODE REVIEW

**Comment by taiki-e** [comment]
> Tested with `cargo run -- https://github.com/openrr/urdf-viz/raw/main/sample.urdf` on macOS.

That file does not contain a mesh, so I don't think this case can be tested.

**Comment by taiki-e** [comment]
I think the URDF used in the gallery in readme is suitable for testing.

e.g., UBR1

```
urdf-viz https://raw.githubusercontent.com/unboundedrobotics/ubr1_preview/6ce886f4d8d02999c442e17f6ee2685fea97b70e/ubr1_description/robots/ubr1_robot.urdf --package-path ubr1_description=https://raw.githubusercontent.com/unboundedrobotics/ubr1_preview/6ce886f4d8d02999c442e17f6ee2685fea97b70e/ubr1_description
```

- Sawyer (Rethink Robotics) https://github.com/RethinkRobotics/sawyer_robot/tree/d1ef3d0ca83d0a943839783efd9b162e6930f2b7/sawyer_description/meshes
- nextage (Kawada) https://github.com/tork-a/rtmros_nextage/tree/ac270fb969fa54abeb6863f9b388a9e20c1f14e0/nextage_description/urdf/meshes
- HSR (Toyota) https://github.com/ToyotaResearchInstitute/hsr_meshes/tree/2d831996c0f9548164a4df5886104793e557897a/meshes
- UBR1 (unbound Robotics) https://github.com/unboundedrobotics/ubr1_preview/tree/6ce886f4d8d02999c442e17f6ee2685fea97b70e/ubr1_description/meshes
- Pepper (Softbank) https://github.com/ros-naoqi/pepper_robot/tree/efad3979b3745d17d5f73d4f97b1fe2b00a8f164/pepper_description (?)
- PR2 (Willow Garage) https://github.com/PR2/pr2_common/tree/b34703bcca2b07cadbc3777d3c504c232a0c0c28/pr2_description/meshes
- thormang3 (Robotis) https://github.com/ROBOTIS-GIT/ROBOTIS-THORMANG-Common/tree/c1159a16dfdca00db1c9b00fed78c2e116e338a9/thormang3_description/meshes

**Comment by H1rono** [comment]
The command you gave me worked with no errors.

File: src/mesh.rs [hunk 1]
```diff
@@ -102,8 +103,19 @@ pub fn load_mesh(
...
+        let buf = if is_url {
+            crate::utils::fetch_read(file_string)?
+        } else {
+            use std::io::Read;
+            let mut file = std::fs::File::open(file_string)?;
+            let mut buf = vec![];
+            file.read_to_end(&mut buf)?;
+            buf
+        };
+        let data = String::from_utf8(buf)?;
```
Comment (taiki-e) [inline src/mesh.rs:117]: Some meshes are not strings (e.g., binary STL), so I think this would not work well with them.

**Review by taiki-e — APPROVED** [review]
Thanks!

**Pull request merged**